packages/monaco/src/browser/textmate/textmate-registry.ts

```diff
@@ -45,8 +46,14 @@ export class TextmateRegistry {
     readonly languageIdToScope = new Map<string, string>();
 
     registerTextmateGrammarScope(scope: string, description: GrammarDefinitionProvider): void {
-        if (this.scopeToProvider.has(scope)) {
-            console.warn(new Error(`a registered grammar provider for '${scope}' scope is overridden`));
+        const existingProvider = this.scopeToProvider.get(scope);
+        if (existingProvider) {
+            Promise.all([existingProvider.getGrammarDefinition(), description.getGrammarDefinition()]).then(([a, b]) => {
+                if (a.location !== b.location) {
```

Comment:
only if both locations are not undefined?

Reply:
Yes, I think that's enough. But if you insist ... :-)